src/OpenTelemetry/Logs/LogRecord.cs

```diff
@@ -81,21 +107,97 @@ public sealed class LogRecord
 
         public EventId EventId { get; }
 
-        public string FormattedMessage { get; }
+        /// <summary>
+        /// Gets or sets the FormattedMessage of the log.
+        /// </summary>
+        public string FormattedMessage
+        {
+            get => this._formattedMessage;
+            set
+            {
+                Guard.ThrowIfNull(value, nameof(this.FormattedMessage));
+                try
+                {
+                    Guard.ThrowIfNullOrEmpty(value, nameof(this.FormattedMessage));
+                    this._formattedMessage = value;
+                } finally
+                {
+                    if (_lockTaken)
+                    {
+                        _spinlock.Exit();
+                    }
+                }
+            }
+        }
 
         /// <summary>
-        /// Gets the raw state attached to the log. Set to <see
+        /// Gets or sets the raw state attached to the log. Set to <see
         /// langword="null"/> when <see
         /// cref="OpenTelemetryLoggerOptions.ParseStateValues"/> is enabled.
         /// </summary>
-        public object State { get; }
+        public object State
+        {
+            get => this._stateList;
```

Comment:
Did not add a lock here, because get is a hot path and if thread A will only get the value before or after the list assignment, which is atomic.